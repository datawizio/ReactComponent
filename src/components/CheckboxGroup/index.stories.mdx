import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import CheckboxGroup from "./index";
import Checkbox from "../Checkbox/index";

<Meta title="Components/Checkbox/CheckboxGroup" component={CheckboxGroup} />

# CheckboxGroup

Компонент `CheckboxGroup` предназначен для вывода группы чекбоксов.

Базовый компонент:
[**antd/checkbox.group**](https://ant.design/components/checkbox/).

## Basic

<Preview>
  <Story name="Basic">
    <CheckboxGroup
      options={[
        { label: "Apple", value: "Apple" },
        { label: "Pear", value: "Pear" },
        { label: "Orange", value: "Orange" }
      ]}
      defaultValue={["Apple"]}
    />
  </Story>
</Preview>

## Disabled

<Preview>
  <Story name="Disabled">
    <CheckboxGroup
      options={[
        { label: "Apple", value: "Apple" },
        { label: "Pear", value: "Pear" },
        { label: "Orange", value: "Orange" }
      ]}
      disabled
      defaultValue={["Apple"]}
    />
  </Story>
</Preview>

## Custom layout

<Preview>
  <Story name="Custom layout">
    <CheckboxGroup defaultValue={["Pear"]}>
      <Checkbox value="Apple">Apple</Checkbox>
      <br />
      <Checkbox value="Pear">Pear</Checkbox>
      <br />
      <Checkbox value="Orange">Orange</Checkbox>
    </CheckboxGroup>
  </Story>
</Preview>

## Props

- [**antd/checkbox.group**](https://ant.design/components/checkbox/#Checkbox-Group)

<Props of={CheckboxGroup} />
