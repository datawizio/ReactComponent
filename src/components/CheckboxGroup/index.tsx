import React from "react";

import { Checkbox } from "antd";
import { CheckboxGroupProps as AntCheckboxGroupProps } from "antd/lib/checkbox";

import "./index.less";

export interface CheckboxGroupProps extends AntCheckboxGroupProps {}

const CheckboxGroup: React.FC<CheckboxGroupProps> = props => {
  return <Checkbox.Group {...props} />;
};

export default CheckboxGroup;
