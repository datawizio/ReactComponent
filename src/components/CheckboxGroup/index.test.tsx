import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import CheckboxGroup from "./index";

const mockProps = {
  className: "customClassName"
};

const setUp = (props?) => mount(<CheckboxGroup {...props} />);

describe("CheckboxGroup component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("CheckboxGroup rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("CheckboxGroup get correct className", () => {
    expect(component.find(`.${mockProps.className}`).length).toBeTruthy();
  });
});
