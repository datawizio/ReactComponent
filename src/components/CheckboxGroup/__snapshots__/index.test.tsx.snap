// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup component CheckboxGroup rendered correctly 1`] = `
<CheckboxGroup
  className="customClassName"
>
  <Memo(CheckboxGroup)
    className="customClassName"
  >
    <div
      className="ant-checkbox-group customClassName"
    />
  </Memo(CheckboxGroup)>
</CheckboxGroup>
`;
